libcontainer/rootfs_linux.go

```diff
@@ -437,9 +488,13 @@ func mountToRootfs(m *configs.Mount, c *mountConfig) error {
 		if err := prepareBindMount(m, rootfs, mountFd); err != nil {
 			return err
 		}
-		if err := mountPropagate(m, rootfs, mountLabel, mountFd); err != nil {
-			return err
+
+		if mountByMove(m, dest) != nil {
+			if err := mountPropagate(m, rootfs, mountLabel, mountFd); err != nil {
+				return err
```

Comment:
This means, if we fail to mount with idmap mounts, it just does a bind-mount?

I'm not sure what is the right behavior, because just mounting and ignoring the mapping for that mount will probably create issues to access the files (if the container running inside a userns).

I'm not sure what the right behavior should be. What do you think? And other maintainers?

Reply:
Thinking a little bit how higher layers will use this, yes, I think we need to error out. At the k8s layer we want to start the container with an id mapped mount or not start it at all. If it is started without the id mapped mounts, the files created will have just random garbage as UID/GID, won't be able to read files... it will be a complete mess.

I think if we just return an error if an id mapped mount was requested and we failed to create it is the right path here.

Also, even if in the future we want to open this door (doubt about it), we can always open a door. But we won't be able to close it if we open it now.

@AlexeyPerevalov Can we make this error out if the id map mount fails?